src/instructlab/config/init.py

```diff
@@ -92,7 +93,12 @@ def init(
         param_source = ctx.parent.parent.get_parameter_source("config_file")
 
     if param_source == click.core.ParameterSource.ENVIRONMENT:
-        model_path, taxonomy_path, taxonomy_base, cfg = get_params_from_env(ctx.obj)
+        try:
+            taxonomy_path, taxonomy_base, cfg = get_params_from_env(
+                interactive, ctx.obj
+            )
+        except click.exceptions.Exit as e:
+            ctx.exit(e.exit_code)
```

Comment:
I believe this is done by the `click` runner when you let the `Exit` bubble up. I don't think this handler is necessary. But perhaps I am wrong?

Reply:
this was existing behavior so i'd prefer to keep it just for ease of merging this before the e2e

Reply:
Sure. I've posted [this](https://github.com/instructlab/instructlab/pull/2017) as a follow-up.